### PR TITLE
Handle boost exception being raised from add_and_parse

### PR DESCRIPTION
**Pull request by jackgerrits**

### CODE REVIEW

File: vowpalwabbit/options_boost_po.cc [hunk 1]
```diff
@@ -11,6 +11,9 @@
 #include <iterator>
 #include <utility>
 
+#include <boost/exception/exception.hpp>
```
Comment (lalo) [inline vowpalwabbit/options_boost_po.cc:14]: I'm assuming none of these files are 1.69 specific?

File: vowpalwabbit/options_boost_po.cc [hunk 2]
```diff
@@ -111,6 +114,13 @@ void options_boost_po::add_and_parse(const option_group_definition& group)
     po::store(parsed_options, vm);
     po::notify(vm);
   }
+// It seems as though Boost 1.69 was introduced in 1.69 and is later started to be thrown out of Boost PO.
```
Comment (olgavrou) [inline vowpalwabbit/options_boost_po.cc:117]: the comment is a bit confusing

**Review by olgavrou — APPROVED** [review]

File: vowpalwabbit/options_boost_po.cc (hunk 2 above)
Comment (jackgerrits) [reply vowpalwabbit/options_boost_po.cc:117]: wow yeah I messed that up, thanks for pointing it out

File: vowpalwabbit/options_boost_po.cc (hunk 1 above)
Comment (jackgerrits) [reply vowpalwabbit/options_boost_po.cc:14]: No I don't believe so

**Pull request merged**